### ISSUE TITLE
False positives for `import/no-extraneous-dependencies` in VSCode (Typescript)

### ISSUE THREAD

**Issue by bukharov** [import-js/eslint-plugin-import#2277]
We have a monorepo with several applications under `applications` folder. Each application is a Typescript project (has its own `tsconfig.json`. The `import/no-extraneous-dependencies` rule triggers when importing a module that has the same name in the other application. However, in the command-line/console there is no such error, so it happens only inside VSCode.

Like on the screenshot below:

<img width="899" alt="Screen Shot 2021-10-27 at 4 50 42 PM" src="https://user-images.githubusercontent.com/49371/139145000-ff663d60-b396-4ee8-9d9f-aa3ac6a93c90.png">

The file is `applications/employer/Application.tsx` and it imports `application/employer/types/store.ts` but somehow it gets confused with `applications/admin/types/store.ts`

Settings:

```
  settings: {
    'import/internal-regex': '^@cwealth',
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
        project: ['applications/*/tsconfig.json'],
      },
    },
    'import/external-module-folders': [
      'node_modules',
      path.resolve(__dirname, 'shared/node_modules'),
    ],
    react: {
      version: '16.13.1',
    },
    jest: {
      version: 26,
    },
  },
```

**Comment by bukharov** [comment]
Actually the command line does report those issues. When I said it doesnt I meant the `webpack` compilation from the command line. If I run `eslint` command on the file it reports those errors:

```
npx eslint applications/employer/Application.tsx 

/Users/deniz/Work/cwealth-frontend-mono/applications/employer/Application.tsx
  2:1  error  There should be no empty line within import group                                                                           import/order
  4:1  error  '@cwealth-frontend/admin' should be listed in the project's dependencies. Run 'npm i -S @cwealth-frontend/admin' to add it  import/no-extraneous-dependencies
  4:1  error  `Application.actions` import should occur before import of `react`                                                          import/order
  5:1  error  '@cwealth-frontend/admin' should be listed in the project's dependencies. Run 'npm i -S @cwealth-frontend/admin' to add it  import/no-extraneous-dependencies
  5:1  error  `components/Routes/Routes` import should occur before import of `react`                                                     import/order

✖ 5 problems (5 errors, 0 warnings)
  3 errors and 0 warnings potentially fixable with the `--fix` option.
```

**Comment by bukharov** [comment]
This looks like an `eslint-import-resolver-typescript` issue.

**Comment by ljharb** [comment]
Webpack shouldn't be running eslint at all; that's a testing concern, not a build concern.
